### ISSUE TITLE
MATLAB syntax bug

### ISSUE THREAD

**Issue by y2-Whyto** [coteditor/CotEditor#1196]
## Environment

- CotEditor: 4.0.6 (465)
- System: macOS 11.5（20G71）
- Language: Chinese (Simplified)

## Short Description
In MATLAB scripts, if there is a symbol of the transpose of a matrix (which is a `'` symbol), then all the code after that symbol is recognized as "strings" and shown in the style of that too, until it comes across another `'` symbol, which causes trouble reading MATLAB scripts for users.

## Steps to Reproduce the Issue

I provide my code here. If you input it into the CotEditor you'll find that at the end of Line 34 there's a transpose symbol, and from Line 35 to the end, all the code is presented as strings. But if you put an apostrophe anywhere in the area where there is a recognition error, the code after that apostrophe will just go back to normal.

Here is my code:
```matlab
close all;
clear;

i1 = imread('L71117027_02720030531_B10.tif');
i2 = imread('L71117027_02720030531_B20.tif');
i3 = imread('L71117027_02720030531_B30.tif');
i4 = imread('L71117027_02720030531_B40.tif');
i5 = imread('L71117027_02720030531_B50.tif');
i(:, :, 1) = i3;
i(:, :, 2) = i2;
i(:, :, 3) = i1;
all(:, :, 1) = i1;
all(:, :, 2) = i2;
all(:, :, 3) = i3;
all(:, :, 4) = i4;
all(:, :, 5) = i5;

red = im2double(i3);
nir = im2double(i4);
ndvi = (nir - red) ./ (nir + red);

veg = zeros(size(ndvi));
veg(ndvi > 0.06) = 1;

%% pca
[r, c, b] = size(all);
all_r = reshape(all, [r * c, b]);
all_r = im2double(all_r);
[coef, score, latent] = pca(all_r);

all_pca = reshape(score, [r, c, b]);
contrib = cumsum(latent)./sum(latent);

score2 = score(:, 1 : 3) * coef(:, 1 : 3)';
all_res = reshape(score, [r, c, b]);


%% figures
figure(1);
imshow(i);

figure(10);
imagesc(ndvi);
colormap gray;

figure(20);
imshow(veg);

figure(100);
subplot(2, 3, 1);
imshow(all(:, :, 1));
subplot(2, 3, 2);
imshow(all(:, :, 2));
subplot(2, 3, 3);
imshow(all(:, :, 3));
subplot(2, 3, 4);
imshow(all(:, :, 4));
subplot(2, 3, 5);
imshow(all(:, :, 5));

figure(200);
subplot(2, 3, 1);
imshow(all_pca(:, :, 1));
subplot(2, 3, 2);
imshow(all_pca(:, :, 2));
subplot(2, 3, 3);
imshow(all_pca(:, :, 3));
subplot(2, 3, 4);
imshow(all_pca(:, :, 4));
subplot(2, 3, 5);
imshow(all_pca(:, :, 5));

figure(300);
subplot(2, 3, 1);
imshow(all_res(:, :, 1));
subplot(2, 3, 2);
imshow(all_res(:, :, 2));
subplot(2, 3, 3);
imshow(all_res(:, :, 3));
subplot(2, 3, 4);
imshow(all_res(:, :, 4));
subplot(2, 3, 5);
imshow(all_res(:, :, 5));
```

## Expected Result
I hope that CotEditor can properly tell the transpose mark from the normal single quotation mark. Thanks.

**Comment by 1024jp** [comment]
I close this issue because of the duplication of the existing issue #697.

**Comment by 1024jp** [comment]
This issue will be addressed in the next major update, version 7.0.0, which is scheduled to be released late 2026.
